File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageInteractor.java
package com.wayfair.themoviedb.feature.homepage;

import com.pubbix.base.BaseInteractor;
import com.wayfair.themoviedb.di.ScreenScope;

import javax.inject.Inject;

@ScreenScope
public class HomepageInteractor extends BaseInteractor<HomepageContract.Repository>
implements HomepageContract.Interactor{
    private HomepageContract.Presenter presenter;
    private HomepageContract.Repository repository;
    @Inject
    public HomepageInteractor(HomepageContract.Repository repository) {
        this.repository = repository;
        this.repository.setInteractor(this);
    }

    @Override
    public void setPresenter(HomepageContract.Presenter presenter) {
        this.presenter = presenter;
    }
}

File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageInteractor.kt
package com.wayfair.themoviedb.feature.homepage

import com.pubbix.base.BaseInteractor
import com.wayfair.themoviedb.di.ScreenScope

import javax.inject.Inject

@ScreenScope
class HomepageInteractor @Inject
constructor(private val repository: HomepageContract.Repository) : BaseInteractor<HomepageContract.Repository>(), HomepageContract.Interactor {
    private var presenter: HomepageContract.Presenter? = null

    init {
        this.repository.setInteractor(this)
    }

    override fun setPresenter(presenter: HomepageContract.Presenter) {
        this.presenter = presenter
    }
}
